apps/encryption/lib/Listeners/UserEventsListener.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Encryption\Listeners;
+
+use OC\Core\Events\BeforePasswordResetEvent;
+use OC\Core\Events\PasswordResetEvent;
+use OCA\Encryption\KeyManager;
+use OCA\Encryption\Services\PassphraseService;
+use OCA\Encryption\Session;
+use OCA\Encryption\Users\Setup;
+use OCA\Encryption\Util;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use OCP\IUser;
+use OCP\IUserManager;
+use OCP\IUserSession;
+use OCP\User\Events\BeforePasswordUpdatedEvent;
+use OCP\User\Events\PasswordUpdatedEvent;
+use OCP\User\Events\UserCreatedEvent;
+use OCP\User\Events\UserDeletedEvent;
+use OCP\User\Events\UserLoggedInEvent;
+use OCP\User\Events\UserLoggedOutEvent;
+
+/**
+ * @template-implements IEventListener<UserCreatedEvent|UserDeletedEvent|UserLoggedInEvent|UserLoggedOutEvent|BeforePasswordUpdatedEvent|PasswordUpdatedEvent|BeforePasswordResetEvent|PasswordResetEvent>
+ */
+class UserEventsListener implements IEventListener {
+
+	public function __construct(
+		private Util $util,
+		private Setup $userSetup,
+		private Session $session,
+		private KeyManager $keyManager,
+		private IUserManager $userManager,
+		private IUserSession $userSession,
+		private PassphraseService $passphraseService,
+	) {
+	}
+
+	public function handle(Event $event): void {
+		if ($event instanceof UserCreatedEvent) {
+			$this->onUserCreated($event->getUid(), $event->getPassword());
+		} elseif ($event instanceof UserDeletedEvent) {
+			$this->onUserDeleted($event->getUid());
+		} elseif ($event instanceof UserLoggedInEvent) {
+			$this->onUserLogin($event->getUid(), $event->getPassword());
+		} elseif($event instanceof UserLoggedOutEvent) {
+			$this->onUserLogout();
+		} elseif($event instanceof BeforePasswordUpdatedEvent) {
+			$this->onBeforePasswordUpdated($event->getUser(), $event->getPassword(), $event->getRecoveryPassword());
+		} elseif($event instanceof PasswordUpdatedEvent) {
+			$this->onPasswordUpdated($event->getUid(), $event->getPassword(), $event->getRecoveryPassword());
+		} elseif($event instanceof BeforePasswordResetEvent) {
+			$this->onBeforePasswordReset($event->getUid());
+		} elseif($event instanceof PasswordResetEvent) {
+			$this->onPasswordReset($event->getUid(), $event->getPassword());
+		}
+	}
+
+	/**
+	 * Startup encryption backend upon user login
+	 *
+	 */
+	private function onUserLogin(string $userId, ?string $password): void {
+		// ensure filesystem is loaded
+		if (!\OC\Files\Filesystem::$loaded) {
+			\OC_Util::setupFS($userId);
+		}
```

Comment:
```suggestion
		$this->setupManager->setupForUser($user);
```
Discussed this with @icewind1991 , it’s safer to keep the call but please use the modern SetupManager class.
You will need the IUser instance from the event.

Reply:
Sadly still no OCP :(